src/Npgsql.GeoJSON/Internal/GeoJSONTypeInfoResolver.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using GeoJSON.Net;
+using GeoJSON.Net.Geometry;
+using Npgsql.Internal;
+using Npgsql.Internal.Postgres;
+
+namespace Npgsql.GeoJSON.Internal;
+
+sealed class GeoJSONTypeInfoResolver : IPgTypeInfoResolver
+{
+    TypeInfoMappingCollection Mappings { get; }
+
+    internal GeoJSONTypeInfoResolver(GeoJSONOptions options, bool geographyAsDefault, CrsMap? crsMap = null)
+    {
+        Mappings = new TypeInfoMappingCollection();
+        AddInfos(Mappings, options, geographyAsDefault, crsMap);
+        // TODO opt-in arrays
```

Comment:
This todo is for the frankenfactory in a later PR right?

Reply:
Yes all resolvers need to be reorganized as part of that work